src/main/java/org/fundacionjala/coding/rodrigo/movierefactor/Rental.java

```diff
@@ -0,0 +1,84 @@
+package org.fundacionjala.coding.rodrigo.movierefactor;
+
+/**
+ * Rental for the movies.
+ */
+public final class Rental {
+    private Customer customer;
+    private String bill;
+    private double totalAmount;
+    private int totalFRP;
+
+    /**
+     * Constructor.
+     *
+     * @param customer Input customer.
+     */
+    Rental(final Customer customer) {
+        this.customer = customer;
+        totalAmount = 0;
+        totalFRP = 0;
+        calculateFrequentRenterPoints();
+        calculateAmount();
+        makeBill();
+    }
+
+    /**
+     * Calculates the total amount.
+     */
+    private void calculateAmount() {
+        for (Movie movie : customer.getMovies()) {
+            totalAmount += movie.getCharge();
+        }
+    }
+
+    /**
+     * Getter for totalAmount.
+     *
+     * @return total.
+     */
+    double getTotalAmount() {
+        return totalAmount;
+    }
+
+    /**
+     * Calculates the Frequent renter points.
+     */
+    private void calculateFrequentRenterPoints() {
+        for (Movie movie : customer.getMovies()) {
+            totalFRP += movie.getFrequentRenterPoints();
+        }
+    }
+
+    /**
+     * Makes the bill for print.
+     */
+    private void makeBill() {
+        StringBuilder bill = new StringBuilder(String.format("Rental Record for: %s %n", customer.getName()));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "bill" which hides the field declared at line 8. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)